table-api/src/main/java/io/deephaven/api/agg/util/AggCountType.java

```diff
@@ -0,0 +1,319 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.api.agg.util;
+
+import io.deephaven.util.QueryConstants;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * The types of counts that can be performed.
+ */
+public enum AggCountType {
+    ALL, NON_NULL, NULL, NEGATIVE, POSITIVE, ZERO, NAN, INFINITE, FINITE;
```

Comment:
I think we need Javadoc on each enum value describing what it does and what it applies to.  For example, I would not expect NAN or INFINITE to make sense for anything but floats.